Source/carthage/swift-is-crashy.c

```diff
@@ -27,10 +28,29 @@ static const char * const exceptionMessage = "\n"
 	"of Carthage and any crash report found in Console.app.\n"
 	"\n";
 
-static void uncaughtSignal(int zig, siginfo_t *info, void *context) {
-	const char *signalName = strsignal(zig);
-	fprintf(stderr, exceptionMessage, signalName);
+// strnlen isn't on the async-signal safe list, but the implementation is 
+// pretty much the same as this. If this is overkill, just replace with 
+// the standard strnlen.
+static size_t safe_strnlen(const char * string, size_t max) {
+	for (size_t i = 0; i < max; ++i) {
+		if (string[i] == '\0') {
+			return i;
+		}
+	}
 
+	return max;
+}
+
+static void uncaughtSignal(int zig, siginfo_t *info, void *context) {
+	size_t preludeLength = safe_strnlen(exceptionPrelude, 90); // 47 at this time
```

Comment:
Maybe add a comment to the top of the file to make it more obvious that these lengths have to always be greater?